src/bun.js/bindings/webcore/JSWasmStreamingCompiler.cpp

```diff
@@ -0,0 +1,247 @@
+    auto& impl = castedThis->wrapped();
+
+    auto chunkValue = callFrame->uncheckedArgument(0);
+    std::span<const uint8_t> chunk;
+
+    // See getWasmBufferFromValue in JSC's JSWebAssemblyHelpers.h
+    if (auto arrayBufferView = jsDynamicCast<JSArrayBufferView*>(chunkValue)) {
+        if (isTypedArrayType(arrayBufferView->type())) {
+            validateTypedArray(lexicalGlobalObject, arrayBufferView);
+            RETURN_IF_EXCEPTION(throwScope, {});
+        } else {
+            // DataView
+            IdempotentArrayBufferByteLengthGetter<std::memory_order_relaxed> getter;
+            if (!jsCast<JSDataView*>(arrayBufferView)->viewByteLength(getter)) [[unlikely]] {
+                throwTypeError(lexicalGlobalObject, throwScope, typedArrayBufferHasBeenDetachedErrorMessage);
+                return {};
+            }
+        }
+
+        chunk = { static_cast<const uint8_t*>(arrayBufferView->vector()), arrayBufferView->byteLength() };
+    } else if (auto arrayBuffer = jsDynamicCast<JSArrayBuffer*>(chunkValue)) {
+        auto arrayBufferImpl = arrayBuffer->impl();
+        if (arrayBufferImpl->isDetached()) {
+            throwTypeError(lexicalGlobalObject, throwScope, typedArrayBufferHasBeenDetachedErrorMessage);
+            return {};
+        }
+
+        chunk = { static_cast<const uint8_t*>(arrayBufferImpl->data()), arrayBufferImpl->byteLength() };
+    } else [[unlikely]] {
+        // See WasmStreamingObject::Push in Node.js's node_wasm_web_api.cc
+        return Bun::ERR::INVALID_ARG_TYPE(throwScope, lexicalGlobalObject, "chunk must be an ArrayBufferView or an ArrayBuffer");
+    }
+
+    RETURN_IF_EXCEPTION(throwScope, {});
+    impl.addBytes(chunk);
+
+    return encodedJSUndefined();
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsWasmStreamingCompilerPrototypeFunction_addBytes, (JSGlobalObject * lexicalGlobalObject, CallFrame* callFrame))
+{
+    return IDLOperation<JSWasmStreamingCompiler>::call<jsWasmStreamingCompilerPrototypeFunction_addBytesBody>(*lexicalGlobalObject, *callFrame, "addBytes");
+}
+
+static inline EncodedJSValue jsWasmStreamingCompilerPrototypeFunction_finalizeBody(JSGlobalObject* lexicalGlobalObject, CallFrame*, typename IDLOperation<JSWasmStreamingCompiler>::ClassParameter castedThis)
+{
+    castedThis->wrapped().finalize(lexicalGlobalObject);
+    return encodedJSUndefined();
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsWasmStreamingCompilerPrototypeFunction_finalize, (JSGlobalObject * lexicalGlobalObject, CallFrame* callFrame))
+{
+    return IDLOperation<JSWasmStreamingCompiler>::call<jsWasmStreamingCompilerPrototypeFunction_finalizeBody>(*lexicalGlobalObject, *callFrame, "finalize");
+}
+
+static inline EncodedJSValue jsWasmStreamingCompilerPrototypeFunction_failBody(JSGlobalObject* lexicalGlobalObject, CallFrame* callFrame, typename IDLOperation<JSWasmStreamingCompiler>::ClassParameter castedThis)
+{
+    auto& vm = JSC::getVM(lexicalGlobalObject);
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
```

Comment:
unused?

Reply:
That's what I deserve for blindly copying from other files :P

I'll remove that.